src/utils/os.c

```diff
@@ -890,6 +890,12 @@ bool signal_process(char *proc_name, int sig) {
     return false;
   }
 
+  if (strcmp(proc_name, ".") == 0) {
+    // may be caused by somebody calling basename(proc_name);
```

Comment:
Nah, I think this is even worse:

> Both dirname() and basename() may modify the contents of path, so
> it may be desirable to pass a copy when calling one of these
> functions.

Reply:
This one of those cases where if the programmers does not read the docs fully all can end in a major disaster:
```verbatim
       In the glibc implementation, the POSIX versions of these
       functions modify the path argument, and segfault when called with
       a static string such as "/usr/".
```